src/_guides/language/sound-problems.md

```diff
@@ -466,29 +486,48 @@ filterValues((x) => (x [!as String!]).contains('Hello'));
 
 <hr>
 
-<a id="common-errors-and-warnings"></a>
-## Runtime errors
+### Incorrect type inference
 
-Dart enforces a sound type system. Roughly, this means
-you can't get into a situation where the value stored in a variable is
-different from the variable's static type. Like most modern statically
-typed languages, Dart accomplishes this with a combination of static
-(compile-time) and dynamic (runtime) checking.
+```nocode
+error - ...
+```
```

Comment:
@mit-mit @kevmoo I would like to add the actual error message that would show up with the example mentioned in the [change log](https://github.com/dart-lang/sdk/blob/master/CHANGELOG.md#language-1) to match the rest of the page. I could not find it though, so I'm wondering the best way to find relevant prs when starting from the change log. 

I followed the link in the change log to the issue, but couldn't locate the actual PR where the error message was added to the code (if that error message was actually added as part of 2.18, as it being mentioned in the change log implies).

Reply:
My recommendation with these is to include the code in a project and then run `dart analyze` to see the resulting error. It's a bit hard to follow the code implemented in the SDK, as errors touch a lot of pieces.

In this case, with the code you included below, I get an error like the following:

```
error - The return type '...' isn't a '...', as required by the closure's context - return_of_invalid_type_from_closure
```

Eventually, the goal is to get the code in to an example(https://github.com/dart-lang/site-www/tree/main/examples/type_system) so we can keep track of the actual error in an analysis results file(https://github.com/dart-lang/site-www/blob/main/examples/type_system/analyzer-results-stable.txt) and use the code excerpt tool when referencing it. That way if the error message updates we'll know to update this. This is a bit of a pain to set up currently though for error messages, especially ones not yet in stable, so it's fine if we add that later.